test/byContract/IexecPoco/02_contribute.test.ts

```diff
@@ -0,0 +1,517 @@
+                }).orders,
+            );
+            await iexecPoco.initialize(dealId, taskIndex).then((tx) => tx.wait());
+            const workerTaskStake = await iexecPoco
+                .viewDeal(dealId)
+                .then((deal) => deal.workerStake.toNumber());
+            const workers = [
+                { wallet: worker1, resultDigest: resultDigest },
+                { wallet: worker2, resultDigest: wrongResultDigest },
+                { wallet: worker3, resultDigest: resultDigest },
+                { wallet: worker4, resultDigest: resultDigest },
+            ];
+            const winningWorkers = [worker1, worker3, worker4];
+            // worker2 is a losing worker
+            let task;
+            let contributeBlockTimestamp;
+            for (let i = 0; i < workers.length; i++) {
+                const worker = workers[i];
+                const workerAddress = worker.wallet.address;
+                const { resultHash, resultSeal } = buildResultHashAndResultSeal(
+                    taskId,
+                    worker.resultDigest,
+                    worker.wallet,
+                );
+                await iexecWrapper.depositInIexecAccount(worker.wallet, workerTaskStake);
+                const viewFrozenOf = (address: string) =>
+                    iexecPoco.frozenOf(address).then((frozen) => frozen.toNumber());
+                const frozenBefore = await viewFrozenOf(workerAddress);
+                contributeBlockTimestamp = await setNextBlockTimestamp();
+                const tx = await iexecPoco
+                    .connect(worker.wallet)
+                    .contribute(
+                        taskId,
+                        resultHash,
+                        resultSeal,
+                        enclave.address,
+                        await buildAndSignPocoClassicEnclaveMessage(
+                            resultHash,
+                            resultSeal,
+                            enclave,
+                        ),
+                        await buildAndSignContributionAuthorizationMessage(
+                            workerAddress,
+                            taskId,
+                            enclave.address,
+                            sms,
+                        ),
+                    );
+                const contribution = await iexecPoco.viewContribution(taskId, workerAddress);
+                expect(contribution.status).equal(ContributionStatusEnum.CONTRIBUTED);
+                expect(contribution.resultHash).equal(resultHash);
+                expect(contribution.resultSeal).equal(resultSeal);
+                expect(contribution.enclaveChallenge).equal(enclave.address);
+                task = await iexecPoco.viewTask(taskId);
+                expect(task.contributors.length).equal(i + 1);
+                expect(task.contributors[i]).equal(workerAddress);
+                await expect(tx)
+                    .to.changeTokenBalances(
```

Comment:
Check events then balances.

Reply:
Trying stick as much as possible to the order of what has been coded in the implem for code review & maintenance..!